### ISSUE TITLE
 Could not find session in local session map.

### ISSUE THREAD

**Issue by google-code-exporter** [magro/memcached-session-manager#207]
_From [huxiaobi...@gmail.com](https://code.google.com/u/105370140000329768710/) on June 07, 2013 09:49:29_

2013-06-07 15:06:40 WARN  de.javakaffee.web.msm.MemcachedSessionService - Could not find session in local session map.
java.lang.IllegalArgumentException: The given node id is not part of this list
	at de.javakaffee.web.msm.NodeIdList.getNextNodeId(NodeIdList.java:60)
	at de.javakaffee.web.msm.MemcachedNodesManager.getNextAvailableNodeId(MemcachedNodesManager.java:351)
	at de.javakaffee.web.msm.MemcachedSessionService.loadBackupSession(MemcachedSessionService.java:904)
	at de.javakaffee.web.msm.MemcachedSessionService.changeSessionIdOnMemcachedFailover(MemcachedSessionService.java:880)
	at de.javakaffee.web.msm.RequestTrackingContextValve.changeRequestedSessionId(RequestTrackingContextValve.java:128)
	at de.javakaffee.web.msm.RequestTrackingContextValve.invoke(RequestTrackingContextValve.java:98)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:127)
	at de.javakaffee.web.msm.RequestTrackingHostValve.invoke(RequestTrackingHostValve.java:165)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:102)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:109)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:293)
	at org.apache.coyote.http11.Http11Processor.process(Http11Processor.java:859)
	at org.apache.coyote.http11.Http11Protocol$Http11ConnectionHandler.process(Http11Protocol.java:602)
	at org.apache.tomcat.util.net.JIoEndpoint$Worker.run(JIoEndpoint.java:489)
	at java.lang.Thread.run(Thread.java:662)

_Original issue: http://code.google.com/p/memcached-session-manager/issues/detail?id=166_

**Comment by google-code-exporter** [comment]
_From [martin.grotzke](https://code.google.com/u/martin.grotzke/) on June 07, 2013 01:40:46_

Can you share more information? E.g. your configuration, more parts of the log that might show which nodeId is requested (there should be s.th. like "Session needs to be relocated as node $nodeId is not available, loading backup session for $requestedSessionId").

**Comment by google-code-exporter** [comment]
_From [huxiaobi...@gmail.com](https://code.google.com/u/105370140000329768710/) on June 07, 2013 20:52:04_

Here is my configuration  <Manager className="de.javakaffee.web.msm.MemcachedBackupSessionManager"
                         memcachedNodes="n1:192.168.3.14:11211,n2:192.168.3.15:11211"
                         sticky="false"
                         operationTimeout="3000"
                         sessionBackupAsync="false"
                         sessionBackupTimeout="1800000"
                         lockingMode="none"
                         requestUriIgnorePattern=".*\.(ico|png|gif|jpg|css|js)$"
                         transcoderFactoryClass="de.javakaffee.web.msm.serializer.kryo.KryoTranscoderFactory"
                /> and there is no such 'Session needs to be relocated as node' info in my log

**Comment by google-code-exporter** [comment]
_From [martin.grotzke](https://code.google.com/u/martin.grotzke/) on June 14, 2013 02:40:25_

Does this occur often, or did it occur only once?
Have you changed your tomcat/msm configuration?
Have you changed your memcached setup, anything in your infrastructure?
So, did it happen during "normal" operation or was anything special?

**Comment by google-code-exporter** [comment]
_From [martin.grotzke](https://code.google.com/u/martin.grotzke/) on June 15, 2013 04:09:01_

Improved logging so that the requested nodeId is logged (and available node ids).

**Comment by google-code-exporter** [comment]
_From [huxiaobi...@gmail.com](https://code.google.com/u/105370140000329768710/) on June 18, 2013 01:40:30_

this occur often when my appliacation run  , i don't change anything about configuration.

**Comment by google-code-exporter** [comment]
_From [martin.grotzke](https://code.google.com/u/martin.grotzke/) on June 18, 2013 08:04:40_

Can you please upgrade to msm 1.6.5, which contains improved logging for this case (prints out the requested node id) and send log output again?
If this does not provide any insight the next step would be to create a sample app that allows me to reproduce your issue (until now, with my samples I can't).

**Comment by google-code-exporter** [comment]
_From [huxiaobi...@gmail.com](https://code.google.com/u/105370140000329768710/) on June 19, 2013 00:49:28_

java.lang.IllegalArgumentException: The given node id lwzjB7t is not part of this list [n1, n2]
	at de.javakaffee.web.msm.NodeIdList.getNextNodeId(NodeIdList.java:60)
	at de.javakaffee.web.msm.MemcachedNodesManager.getNextAvailableNodeId(MemcachedNodesManager.java:351)
	at de.javakaffee.web.msm.MemcachedSessionService.loadBackupSession(MemcachedSessionService.java:921)
	at de.javakaffee.web.msm.MemcachedSessionService.changeSessionIdOnMemcachedFailover(MemcachedSessionService.java:897)
	at de.javakaffee.web.msm.RequestTrackingContextValve.changeRequestedSessionId(RequestTrackingContextValve.java:128)
	at de.javakaffee.web.msm.RequestTrackingContextValve.invoke(RequestTrackingContextValve.java:98)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:127)
	at de.javakaffee.web.msm.RequestTrackingHostValve.invoke(RequestTrackingHostValve.java:170)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:102)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:109)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:298)
	at org.apache.coyote.http11.Http11Processor.process(Http11Processor.java:852)
	at org.apache.coyote.http11.Http11Protocol$Http11ConnectionHandler.process(Http11Protocol.java:588)
	at org.apache.tomcat.util.net.JIoEndpoint$Worker.run(JIoEndpoint.java:489)
	at java.lang.Thread.run(Thread.java:662)
new issue happen when i change 1.6.5

**Comment by google-code-exporter** [comment]
_From [martin.grotzke](https://code.google.com/u/martin.grotzke/) on June 19, 2013 02:39:53_

Doesn't help much. Can you configure logging according to <a href="https://code.google.com/p/memcached-session-manager/wiki/SetupAndConfiguration#Configure_logging" rel="nofollow">https://code.google.com/p/memcached-session-manager/wiki/SetupAndConfiguration#Configure_logging</a> and send log output? Especially lines containing "&gt;&gt;&gt;&gt;&gt;&gt; Request starting:" are of interest (and those that contain a node id from "The given node id ... is not part of this list"). I'm wondering what session ids are sent by your clients.

**Comment by google-code-exporter** [comment]
_From [martin.grotzke](https://code.google.com/u/martin.grotzke/) on August 24, 2013 15:24:24_

Any feedback?

**Comment by google-code-exporter** [comment]
_From [JamesGos...@gmail.com](https://code.google.com/u/101338783126219327492/) on June 25, 2015 23:08:46_

Yes，I have encountered the same issue！Does anyone who know how to fix it？

**Comment by zhipingch** [comment]
Yes，I have encountered the same issue！Does anyone who know how to fix it？
